File: java/socketd/src/main/java/org/noear/socketd/transport/core/BufferWriter.java
package org.noear.socketd.transport.core;

import java.io.IOException;

/**
 * @author noear
 * @since 2.0
 */
public interface BufferWriter {
    void putBytes(byte[] bytes) throws IOException;
    void putBytes(byte[] src, int offset, int length) throws IOException;

    void putInt(int val) throws IOException;
    void putChar(int val) throws IOException;
    void flush() throws IOException;
}

File: javascript/ts/org/noear/socketd/transport/core/buffer/BufferWriter.ts
/**
 * 缓冲写
 *
 * @author noear
 * @since 2.0
 * @class
 */
export interface BufferWriter {
    /**
     * 推入一组 byte
     * @param {byte[]} src
     * @param {number} offset
     * @param {number} length
     */
    putBytes(src: ArrayBuffer, offset: number, length: number);

    /**
     * 推入 int
     * @param {number} val
     */
    putInt(val: number);

    /**
     * 推入 char
     * @param {number} val
     */
    putChar(val: number);

    /**
     * 冲刷
     */
    flush();
}